### DISCUSSION TITLE
Avoid compiling unnecessary classes / code to reduce bundle size

### DISCUSSION THREAD

**Discussion by bfelbo** [jrouwe/JoltPhysics#1146] [Q&A] [answered]
I'm trying to use Jolt Physics for HTML5 games. For such games, the amount of code being bundled is crucial for the game to download and load faster. Especially old devices can take multiple seconds just to evaluate all the Jolt Physics code.

Game engines like Godot have the same problem and have implemented [Feature Build Profiles](https://github.com/godotengine/godot/pull/62996) so that you can avoid bundling unnecessary classes etc. Does Jolt have a similar way to avoid compiling parts of the code?

**Comment by jrouwe** [comment, marked as answer]
Some effort has been done to reduce the library size, see:

https://github.com/jrouwe/JoltPhysics.js/discussions/164

But there are no feature flags and when I looked at it there were also no obvious ways to save a lot of memory either.

**Comment by bfelbo** [comment]
Thanks for the quick response. That makes sense.

What if you only need Jolt Physics for 2D? Would it be possible to save space/memory by not including the 3D parts?

**Reply by jrouwe** [reply]
The 2D implementation is just the 3D simulation with 1 axis locked. So no, that wouldn't save anything.

**Comment by bfelbo** [comment]
Makes sense. Closing this to avoid fragmenting the discussion. https://github.com/jrouwe/JoltPhysics.js/discussions/164 seems like the best place to continue any discussion on how to reduce size / memory usage.
